### PR TITLE
Refactor metrics handler

### PR DESCRIPTION
**Pull request by pdewilde**

### CODE REVIEW

**Review by sailorlqh — APPROVED** [review]

File: cmd/main.go [hunk 1]
```diff
@@ -128,14 +86,15 @@ func realMain(ctx context.Context) error {
...
+				if err = db.Update(ctx, dbUpdateParams); err != nil {
+					logger.WarnContext(ctx, "Error updating metrics definitions, will use cached definition if available.", "err", err.Error())
+				}
 			}
 		}
 	}()
 
 	mux := http.NewServeMux()
-	mux.Handle("POST /sendMetrics", handleMetric(h, db))
+	mux.Handle("POST /sendMetrics", server.HandleMetric(h, db))
```
Comment (sailorlqh) [inline cmd/main.go:97]: Wasn't aware we can specify POST this way

Comment (pdewilde) [reply cmd/main.go:97]: It's new in https://go.dev/blog/go1.22

Comment (verbanicm) [reply cmd/main.go:97]: https://pkg.go.dev/net/http#hdr-Patterns

Comment (sailorlqh) [reply cmd/main.go:97]: Thanks!

**Pull request merged**